independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/RestResponse.java

```diff
@@ -0,0 +1,2136 @@
+     * @since 2.0
+     */
+    public abstract StatusType getStatusInfo();
+
+    /**
+     * Get the message entity Java instance. Returns {@code null} if the message
+     * does not contain an entity body.
+     * <p>
+     * If the entity is represented by an un-consumed {@link InputStream input stream}
+     * the method will return the input stream.
+     * </p>
+     *
+     * @return the message entity or {@code null} if message does not contain an
+     *         entity body (i.e. when {@link #hasEntity()} returns {@code false}).
+     * @throws IllegalStateException if the entity was previously fully consumed
+     *         as an {@link InputStream input stream}, or
+     *         if the response has been {@link #close() closed}.
+     */
+    public abstract T getEntity();
+
+    /**
+     * Read the message entity input stream as an instance of specified Java type
+     * using a {@link javax.ws.rs.ext.MessageBodyReader} that supports mapping the
+     * message entity stream onto the requested type.
+     * <p>
+     * Method throws an {@link ProcessingException} if the content of the
+     * message cannot be mapped to an entity of the requested type and
+     * {@link IllegalStateException} in case the entity is not backed by an input
+     * stream or if the original entity input stream has already been consumed
+     * without {@link #bufferEntity() buffering} the entity data prior consuming.
+     * </p>
+     * <p>
+     * A message instance returned from this method will be cached for
+     * subsequent retrievals via {@link #getEntity()}. Unless the supplied entity
+     * type is an {@link java.io.InputStream input stream}, this method automatically
+     * {@link #close() closes} the an unconsumed original response entity data stream
+     * if open. In case the entity data has been buffered, the buffer will be reset
+     * prior consuming the buffered data to enable subsequent invocations of
+     * {@code readEntity(...)} methods on this response.
+     * </p>
+     *
+     * @param <OtherT> entity instance Java type.
+     * @param entityType the type of entity.
+     * @return the message entity; for a zero-length response entities returns a corresponding
+     *         Java object that represents zero-length data. In case no zero-length representation
+     *         is defined for the Java type, a {@link ProcessingException} wrapping the
+     *         underlying {@link NoContentException} is thrown.
+     * @throws ProcessingException if the content of the message cannot be
+     *         mapped to an entity of the requested type.
+     * @throws IllegalStateException if the entity is not backed by an input stream,
+     *         the response has been {@link #close() closed} already,
+     *         or if the entity input stream has been fully consumed already and has
+     *         not been buffered prior consuming.
+     * @see javax.ws.rs.ext.MessageBodyReader
+     * @since 2.0
```

Comment:
I think we should get rid of all these since tags

Reply:
True. this is less contentious.

Reply:
Done.